feat: add `get_metrics` function in SDK

alvarobartt opened the pull request:
# Description

This PR adds a function in the SDK named `get_metrics` that calls `GET /api/v1/me/datasets/{dataset_id}/metrics` to get the metrics of a certain dataset from a certain user, meaning that each user would just see what can be seen according to its permissions.

This PR is created on top of #3465 since we will need to get the `records.count` to override the magic method `__len__` in `_ArgillaFeedbackDataset` so that the length of a dataset is directly retrieved from the record count of the dataset, since we're no longer keeping local data when working with a remote dataset.

**Type of change**

- [X] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

- [X] Add unit tests for the `get_metrics` function

**Checklist**

- [ ] I added relevant documentation
- [X] follows the style guidelines of this project
- [X] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [x] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

alvarobartt committed:
feat: add `FeedbackMetricsModel`

alvarobartt committed:
feat: add `get_metrics` in SDK

alvarobartt committed:
test: add `get_metrics` & `FeedbackMetricsModel`

alvarobartt committed:
docs: update `CHANGELOG.md`

frascuchon commented on src/argilla/client/sdk/v1/datasets/models.py:
```diff
@@ -100,3 +100,19 @@ class FeedbackSuggestionModel(BaseModel):
+    count: int
+    submitted: int
+    discarded: int
+    draft: int
```
I think we don't need these metrics attributes in the Python client for now. Or do we show the user-connected metrics somewhere?

alvarobartt replied:
It was just to align both schemas, but for now I guess we'll just use `records.count` from the Python client as part of `len(dataset.records)`

gabrielmbmb reviewed (approved):
LGTM

alvarobartt committed:
revert: remove `FeedbackResponsesMetricsModel`

Since it's not used in the Python client, this has been removed, to avoid modifying the model on API changes on fields that are not required

Co-authored-by: Francisco Aranda <francis@argilla.io>

alvarobartt committed:
revert(test): `test_feedback_metrics_schema`
